File: App016_ContactManager2/app/src/main/java/com/example/app016_contactmanager2/db/ContactDAO.java
package com.example.app016_contactmanager2.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.app016_contactmanager2.db.entities.Contact;

import java.util.List;

@Dao
public interface ContactDAO {

    @Insert
    long insertContact(Contact contact);

    @Delete
    void deleteContact(Contact contact);

    @Update
    void updateContact(Contact contact);

    @Query("SELECT * FROM contact_table ORDER BY id DESC")
    List<Contact> getAllContacts();

    @Query("SELECT * FROM contact_table WHERE id == :contactID")
    Contact getContact(long contactID);
}

File: App037_Kt_ContactManager/app/src/main/java/com/kes/app037_kt_contactmanager/room/ContactDAO.kt
package com.kes.app037_kt_contactmanager.room

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ContactDAO {

    @Insert
    suspend fun insertContact(contact: Contact): Long

    @Update
    suspend fun updateContact(contact: Contact)

    @Delete
    suspend fun deleteContact(contact: Contact)

    @Query("DELETE FROM contact")
    suspend fun deleteAll()

    @Query("SELECT * FROM contact")
    fun getAllContacts(): LiveData<List<Contact>>

}